autopush/web/registration.py

```diff
@@ -307,6 +336,15 @@ def _return_endpoint(self, endpoint_data, new_uaid, router=None):
                        client_info=self._client_info)
         self.finish()
 
+    def _list_channels(self):
+        _, channels = self.ap_settings.message.all_channels(self.uaid)
+        dashed = [str(uuid.UUID(x)) for x in channels]
+        self.write(json.dumps(
```

Comment:
_list_channels is called in a separate thread so you really want to go back to the event loop for write/finish